### PR TITLE
fix: show bacalhau id error

### PR DESCRIPTION
**Pull request by hunjixin**
### Review Type Requested (choose one):
- [ ] **Glance** - superficial check (from domain experts)
- [ ] **Logic** - thorough check (from everybody doing review)

show error in "bacalhau id" command.

### Summary
Provide a one line summary and link to any relevant references

### Task/Issue reference

Re: https://github.com/Lilypad-Tech/hni/issues/7

### Details (optional)
Add any additional details that might help Code Reviewers digest this PR

### CODE REVIEW

**Review by AquiGorka — COMMENTED** [review]
@hunjixin thank you for the changes, can you provide a video recording of how this changes affect the POW process?

**Comment by hunjixin** [comment]
bacalhau id have error
image

and when runing rp, lilypad not print pow worker message
image

this pr return this error, and tell user you got sometheing error
image

@AquiGorka

**Comment by hunjixin** [comment]
image

add a new commit to resolve this problem,  cmd output some odd error log. just parse the last line  @AquiGorka

File: pkg/executor/bacalhau/bacalhau.go [hunk 1]
```diff
@@ -49,18 +49,20 @@ func (executor *BacalhauExecutor) Id() (string, error) {
...
 	nodeIdCmd.Env = executor.bacalhauEnv
 
-	output, err := nodeIdCmd.CombinedOutput()
+	runOutputRaw, err := nodeIdCmd.CombinedOutput()
 	if err != nil {
 		return "", fmt.Errorf("error calling get id results %s", err.Error())
 	}
 
+	splitOutputs := strings.Split(strings.Trim(string(runOutputRaw), " \t\n"), "\n")
+	runOutput := splitOutputs[len(splitOutputs)-1]
```
Comment (AquiGorka) [inline pkg/executor/bacalhau/bacalhau.go:58]:
```go
errOutput = splitOutputs[1:]
if errOutput != nil {
  return "", fmt.Errorf("error unmarshalling job JSON %s %s", err.Error(), errOutput)
}
```

Comment (hunjixin) [reply pkg/executor/bacalhau/bacalhau.go:58]: but in correct situation,  [1:] may panic
image

Comment (hunjixin) [reply pkg/executor/bacalhau/bacalhau.go:58]: image
add like this?, but i got a panic in this change. i dont know how the error log look like

Comment (AquiGorka) [reply pkg/executor/bacalhau/bacalhau.go:58]: If you are getting errors, it means there is something wrong, you also need to trim, I added a similar change recently, here: https://github.com/Lilypad-Tech/lilypad/pull/175/files
Let's make sure we also capture errors if there are errors

Comment (walkah) [reply pkg/executor/bacalhau/bacalhau.go:58]: @AquiGorka I've updated to add logging for error output.
@hunjixin Looks like we want `splitOutputs[:len(splitOutputs)-1]` (i.e. everything but the last line).

**Comment by walkah** [comment]
In discussing with @taoshengshi - if we can get this merged, he can do testing in the morning (china time) with RPs to confirm this unblocks the issue.

**Review by AquiGorka — APPROVED** [review]

File: pkg/executor/bacalhau/bacalhau.go (hunk 1 above)
Comment (hunjixin) [reply pkg/executor/bacalhau/bacalhau.go:58]: this change seem to make things not work.the output in my comment 
1 line: error log
1 line: id result
1 line: empty line

this is why add a trim to make sure the empty line is removed

split by \n,  4 lines, with  2 empty line in last

**Pull request merged**